pkg/northbound/gnmi/subscribe.go

```diff
@@ -44,79 +52,105 @@ func (s *Server) Subscribe(stream gnmi.GNMI_SubscribeServer) error {
 	changesChan, err := mgr.Dispatcher.RegisterNbi(hash)
 	if err != nil {
 		log.Println("Subscription present: ", err)
-		return err
+		return status.Error(codes.Internal, err.Error())
 	}
 	opStateChan, err := mgr.Dispatcher.RegisterOpState(hash)
 	if err != nil {
 		log.Println("Subscription present: ", err)
-		return err
+		return status.Error(codes.Internal, err.Error())
 	}
+	resChan := make(chan result)
 	//Handles each subscribe request coming into the server, blocks until a new request or an error comes in
-	for {
-		in, err := stream.Recv()
-		if err == io.EOF {
-			log.Println("Subscription Terminated")
-			//Ignoring Errors during removal
-			mgr.Dispatcher.UnregisterNbi(hash)
-			mgr.Dispatcher.UnregisterOperationalState(hash)
-			return nil
-		}
+	go func() {
```

Comment:
why is this now a go routine - I thought the for was to block it from returning - also I don't like anon go functions like this - could you make it a named function

Reply:
the blocking is done in the for under this function that waits for this to return.
Moved to function.